conbench/util.py

```diff
@@ -109,6 +110,34 @@ def _convert(s: str):
     return [_convert(s) for s in input]
 
 
+def dedent_rejoin(s: str):
+    """
+    Remove common leading whitespace, replace newlines by spaces.
+
+    Useful for being able to write marshmallow property docstrings with
+    indented block paragraphs.
+    """
+    return " ".join(textwrap.dedent(s).strip().splitlines())
```

Comment:
A note on this. For many years I have `Alt +Q` in my editor bound to doing `rewrap` (rewrap text so that it nicely fills up to N columns). The triple-quoted strings in Python allow for using that. But they contain literal newlines, and if indention is applied that there also is some leading whitespace. Those additional newlines and the indention can of course be removed with programmatic tools.

Reply:
Luckily I think Swagger (like Sphinx) automatically collapses all these newlines and indentations before rendering docstrings into HTML. That's why I was able to get away with this triple-quoted string:
https://github.com/conbench/conbench/blob/e9fae1e603b28508cb79c2911e1c525fbf4ee23b/conbench/entities/benchmark_result.py#L363
without having to collapse things myself. It does insert the whitespace garbage into the OpenAPI json, but since that garbage is gone by the time it gets to Swagger I thought it wasn't worth cleaning up.

Reply:
Oh, thanks for that feedback. I didn't reply before. Personally, I like doing the explicit more-or-less tidy constructions of strings in Python, but yeah it's good to know that the Swagger toolchain is liberal with the input 'quality'.